### PR TITLE
Ee 6825 use component trace header

### PR DESCRIPTION
**Pull request by owainrobertsbjss**
Change smoke tests to work by inspecting the x-component-trace header, rather than relying on the HTTP response code.

As smoke tests aren't being used yet I will merge once approved.

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/pttg/testrunner/ComponentHeaderCheckerTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import org.junit.Test;
+
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ComponentHeaderCheckerTest {
```
Comment (Tim-Hammonds) [inline src/test/java/uk/gov/digital/ho/pttg/testrunner/ComponentHeaderCheckerTest.java:12]: Could add a test to show that whitespace in a single list of multiple headers doesn't matter.

**Review by mikehalmaho — APPROVED** [review]

**Review by Tim-Hammonds — APPROVED** [review]

File: src/test/java/uk/gov/digital/ho/pttg/testrunner/ComponentHeaderCheckerTest.java (hunk 1 above)
Comment (owainrobertsbjss) [reply src/test/java/uk/gov/digital/ho/pttg/testrunner/ComponentHeaderCheckerTest.java:12]: Good idea, I'll add it.
